Gazebo9/Gazebo11 build error on armhf: multiple definition of `typeinfo name for SingletonT<gazebo::common::SystemPaths>'

**Environment：**
Linux pi-desktop 5.4.0-1058-raspi #65-Ubuntu SMP PREEMPT Fri Mar 25 12:34:25 UTC 2022 armv7l armv7l armv7l GNU/Linux

**Build package**: gazebo-gazebo9_9.19.0

**Build Error**:
```
[ 18%] Linking CXX shared library libgazebo_common.so
/usr/bin/ld: CMakeFiles/gazebo_common.dir/ColladaLoader.cc.o:(.rodata+0x4): multiple definition of `typeinfo name for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.rodata+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/ColladaLoader.cc.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.data.rel.ro+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/CommonIface.cc.o:(.rodata+0x0): multiple definition of `typeinfo name for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.rodata+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/CommonIface.cc.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.data.rel.ro+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/Material.cc.o:(.rodata+0x0): multiple definition of `typeinfo name for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.rodata+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/Material.cc.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.data.rel.ro+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/ModelDatabase.cc.o:(.rodata+0x4): multiple definition of `typeinfo name for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.rodata+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/ModelDatabase.cc.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.data.rel.ro+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/SystemPaths.cc.o:(.rodata+0x0): multiple definition of `typeinfo name for SingletonT<gazebo::common::ModelDatabase>'; CMakeFiles/gazebo_common.dir/ModelDatabase.cc.o:(.rodata+0x30): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/SystemPaths.cc.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for SingletonT<gazebo::common::ModelDatabase>'; CMakeFiles/gazebo_common.dir/ModelDatabase.cc.o:(.data.rel.ro+0x8): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/SystemPaths.cc.o:(.rodata+0x30): multiple definition of `typeinfo name for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.rodata+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/SystemPaths.cc.o:(.data.rel.ro+0x8): multiple definition of `typeinfo for SingletonT<gazebo::common::SystemPaths>'; CMakeFiles/gazebo_common.dir/BVHLoader.cc.o:(.data.rel.ro+0x0): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/MeshCSG.cc.o:(.rodata+0x1c): multiple definition of `typeinfo name for SingletonT<gazebo::common::MeshManager>'; CMakeFiles/gazebo_common.dir/MeshManager.cc.o:(.rodata+0x1fc): first defined here
/usr/bin/ld: CMakeFiles/gazebo_common.dir/MeshCSG.cc.o:(.data.rel.ro+0x8): multiple definition of `typeinfo for SingletonT<gazebo::common::MeshManager>'; CMakeFiles/gazebo_common.dir/MeshManager.cc.o:(.data.rel.ro+0x0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [gazebo/common/CMakeFiles/gazebo_common.dir/build.make:805: gazebo/common/libgazebo_common.so.9.19.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:3161: gazebo/common/CMakeFiles/gazebo_common.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

--- a/gazebo/common/SingletonT.hh
+++ b/gazebo/common/SingletonT.hh
@@ -52,23 +52,5 @@ class SingletonT
            }
 };
 
-/// \brief Helper to declare typed SingletonT
-// clang doesn't compile if it explicitly specializes a type before
-// the type is defined. (forward declaration is not enough.)
-#ifdef __clang__
 #define GZ_SINGLETON_DECLARE(visibility, n1, n2, singletonType)
-#else
-#define GZ_SINGLETON_DECLARE(visibility, n1, n2, singletonType) \
-namespace n1 \
-{ \
-  namespace n2 \
-  { \
-    class singletonType; \
-  } \
-} \
-template class visibility ::SingletonT<n1::n2::singletonType>;
-#endif
-
-/// \}
-
 #endif

